lgc/state/PalMetadata.cpp

```diff
@@ -969,3 +920,179 @@ void PalMetadata::eraseFragmentInputInfo() {
+  const unsigned numberOfShaderStages = 5;
+  static const std::pair<unsigned, unsigned> shaderTable[numberOfShaderStages] = {
+      {mmSPI_SHADER_PGM_RSRC1_LS, CallingConv::AMDGPU_LS},
+      {mmSPI_SHADER_PGM_RSRC1_HS, CallingConv::AMDGPU_HS},
+      {mmSPI_SHADER_PGM_RSRC1_ES, CallingConv::AMDGPU_ES},
+      {mmSPI_SHADER_PGM_RSRC1_GS, CallingConv::AMDGPU_GS},
+      {mmSPI_SHADER_PGM_RSRC1_VS, CallingConv::AMDGPU_VS}};
+
+  for (unsigned i = 0; i < numberOfShaderStages; ++i) {
+    auto entry = m_registers.find(m_document->getNode(shaderTable[i].first));
+    if (entry != m_registers.end())
+      return shaderTable[i].second;
+  }
+  return CallingConv::AMDGPU_CS;
+}
+
+// =====================================================================================================================
+// Returns the offset of the  first user data register for the given calling convention.
+//
+// @param callingConv : The calling convention
+unsigned PalMetadata::getFirstUserDataReg(unsigned int callingConv) {
+  static const std::map<unsigned, unsigned> shaderTable = {{CallingConv::AMDGPU_LS, mmSPI_SHADER_USER_DATA_LS_0},
+                                                           {CallingConv::AMDGPU_HS, mmSPI_SHADER_USER_DATA_HS_0},
+                                                           {CallingConv::AMDGPU_ES, mmSPI_SHADER_USER_DATA_ES_0},
+                                                           {CallingConv::AMDGPU_GS, mmSPI_SHADER_USER_DATA_GS_0},
+                                                           {CallingConv::AMDGPU_VS, mmSPI_SHADER_USER_DATA_VS_0}};
+  static const std::map<unsigned, unsigned> shaderTableGfx9 = {{CallingConv::AMDGPU_LS, mmSPI_SHADER_USER_DATA_LS_0},
+                                                               {CallingConv::AMDGPU_HS, mmSPI_SHADER_USER_DATA_LS_0},
+                                                               {CallingConv::AMDGPU_ES, mmSPI_SHADER_USER_DATA_ES_0},
+                                                               {CallingConv::AMDGPU_GS, mmSPI_SHADER_USER_DATA_ES_0},
+                                                               {CallingConv::AMDGPU_VS, mmSPI_SHADER_USER_DATA_VS_0}};
+
+  bool isGfx9 = m_pipelineState->getTargetInfo().getGfxIpVersion().major == 9;
+  const std::map<unsigned, unsigned> &currentShaderTable = (isGfx9 ? shaderTableGfx9 : shaderTable);
+  return currentShaderTable.at(callingConv);
+}
+
+// =====================================================================================================================
+// Returns to number of sgprs before the sgprs used for other purposes before the first user data sgps.
+//
+// @param conv : The calling convention of the shader
+unsigned PalMetadata::getNumberOfSgprsBeforeUserData(unsigned int conv) {
+  // GFX9+ merged shader have an extra 8 SGPRs before user data.
+  if (conv != CallingConv::AMDGPU_VS && m_pipelineState->getTargetInfo().getGfxIpVersion().major >= 9)
```

Comment:
nit: I realize this is an existing code, but I think the current convention is to compare using the `GfxIp` object, i.e., `....getGfxIpVersion() >= GfxIp{9, 0, 0}`

Reply:
done